### PR TITLE
feat(app marketplace): add more and collapse 

### PR DESCRIPTION
**Pull request by lavanya-bmw**
## Description

Add more and collapse button to category view

## Why

To improve the 'more' button functionality to category view in app marketplace

## Issue

https://github.com/eclipse-tractusx/portal-frontend/issues/688

## Checklist

Please delete options that are not relevant.

- [x] I have followed the [contributing guidelines](https://github.com/eclipse-tractusx/portal-assets/blob/main/docs/developer/Technical%20Documentation/Dev%20Process/How%20to%20contribute.md#commit-and-pr-guidelines)
- [x] I have performed a self-review of my own code
- [x] I have successfully tested my changes locally

### CODE REVIEW

File: src/components/pages/AppMarketplace/components/AppListGroup/index.tsx [hunk 1]
```diff
@@ -32,26 +32,39 @@ export const AppListGroup = ({
...
-    setItemsShown((prevState) => {
-      return (Number(prevState) + 4).toString()
+    setItemsShown(() => {
+      return items.length.toString()
+    })
+  }
+
+  const collapseItemsShown = () => {
+    setItemsShown(() => {
+      return (4).toString()
```
Comment (manojava-gk) [inline src/components/pages/AppMarketplace/components/AppListGroup/index.tsx:45]: Can use one method to handle this??
```
setItemsShown(() => {
return itemsToShow.length >= items.length ? (4).toString() : items.length.toString()
})
```

File: src/components/pages/AppMarketplace/components/AppListGroup/index.tsx [hunk 2]
```diff
@@ -32,26 +32,39 @@ export const AppListGroup = ({
...
         categoryName={t(
           'content.appstore.appOverviewSection.categories.' +
             category.replace(/\s+/g, '').toLowerCase()
         )}
-        disabled={itemsToShow.length >= items.length}
-        onButtonClick={increaseItemsShown}
+        disabled={items.length <= 4}
+        onButtonClick={
+          itemsToShow.length >= items.length
+            ? collapseItemsShown
```
Comment (manojava-gk) [inline src/components/pages/AppMarketplace/components/AppListGroup/index.tsx:65]: Same as above

**Review by manojava-gk — COMMENTED** [review]
small comment. Rest looks fine

File: src/components/pages/AppMarketplace/components/AppListGroup/index.tsx (hunk 2 above)
Comment (lavanya-bmw) [reply src/components/pages/AppMarketplace/components/AppListGroup/index.tsx]: Done

File: src/assets/locales/de/main.json [hunk 3]
```diff
@@ -1997,6 +1997,7 @@
       "decline": "Decline",
       "confirm": "Bestätigen",
       "more": "Mehr",
+      "collapse": "Zusammenbruch",
```
Comment (oyo) [inline src/assets/locales/de/main.json:2000]: haha - that's an ambiguous translation. In this context I guess the correct german term is "Einklappen"

File: src/components/pages/AppMarketplace/components/AppListGroup/index.tsx [hunk 4]
```diff
@@ -32,26 +32,34 @@ export const AppListGroup = ({
...
+  const handleMoreAndCollapse = () => {
+    if (itemsToShow.length >= items.length) {
+      setItemsShown(() => {
+        return (4).toString()
+      })
+    } else
+      setItemsShown(() => {
+        return items.length.toString()
+      })
   }
```
Comment (oyo) [inline src/components/pages/AppMarketplace/components/AppListGroup/index.tsx:46]: First of all why is `itemsToShow` a string in the old code? That doesn't make any sense and should be fixed here to number. Then the entire function can be written more readable as one line (maybe prettier might reformat):
` const handleMoreAndCollapse = () => setItemsShow(itemsToShow.length >= items.length ? 4 : items.length `

**Review by oyo — CHANGES_REQUESTED** [review]

**Review by oyo — DISMISSED** [review]

**Review by evegufy — DISMISSED** [review]

**Comment by evegufy** [comment]
@jjeroch Could you please confirm that this PR should merged to 24.05 anymore but to 24.08 according to the milestone on https://github.com/eclipse-tractusx/portal-frontend/issues/688?

**Comment by jjeroch** [comment]
@evegufy  & @lavanya-bmw  we should add this to 24.08. please. This is not mission critical for 24.05.

**Comment by evegufy** [comment]
Merge to dev once 2.0.0-RC4 was merged dev

**Review by evegufy — APPROVED** [review]

**Pull request merged**